src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -147,15 +157,27 @@ protected override void ApplyResponseChallenge()
                 properties = new AuthenticationProperties(ChallengeContext.Properties);
             }
 
-            if (string.IsNullOrEmpty(properties.RedirectUri))
+            string redirectToUse = null;
```

Comment:
I'm not sure why one would override the `redirect_uri` parameter from the challenge context (it can be changed from `RedirectToIdentityProviderNotification`), but if you specify an explicit value for `Options.RedirectUri`, the OIDC handler will only listen on this path and ignore the values you could use in `AuthenticationProperties`, no?